src/ax/endpoint/endpoint.go

```diff
@@ -20,12 +20,18 @@ type Endpoint struct {
 }
 
 // NewSender returns an ax.Sender that can be used to send messages from this endpoint.
-func (ep *Endpoint) NewSender(ctx context.Context) (ax.Sender, error) {
+func (ep *Endpoint) NewSender(
+	ctx context.Context,
+	validators []Validator,
+) (ax.Sender, error) {
```

Comment:
Rather than ask the caller for the validators, `ep` should just have a `Validators []Validator` field so that it can be pre-configured.  This same slice would be used by the dispatcher.